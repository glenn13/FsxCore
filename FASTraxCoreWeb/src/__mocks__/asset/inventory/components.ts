import {AxiosResponse} from 'axios';
import {QueryResult} from 'react-query';
import {Component, newComponent} from '../../../entities/asset/inventory/Component';

export const mockComponentsQuery = (): QueryResult<AxiosResponse<Component[]>> =>
  ({data: {data: [newComponent(), newComponent(), newComponent()]}} as QueryResult<
    AxiosResponse<Component[]>
  >);
