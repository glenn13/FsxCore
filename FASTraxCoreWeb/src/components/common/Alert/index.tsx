export {default as Toast} from './Toast';
export {default as Confirm} from './Confirm';
