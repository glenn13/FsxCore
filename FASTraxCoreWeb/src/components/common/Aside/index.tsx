import React from 'react';
import styled from 'styled-components';
import clx from 'classnames';
import {useAppStore} from '@app/providers/app.store';

interface IProps {
  position?: 'right' | 'left';
  width?: Number;
  className?: string;
  flat?: boolean;
}

const AsideLayout = styled.aside<IProps>`
  position: relative;
  z-index: 1;
  display: flex;
  background: ${props => props.theme.sidebar.bgColor};
  ${props => (props.flat ? `display: flex; flex: 1;` : `height: 100vh;`)}
  width: ${props => (props.width ? props.width + 'px' : '15em')};
  max-width: 300px;
  order: ${props => (props.position === 'right' ? 1 : 0)};
  flex-direction: column;
  box-shadow: ${props => (props.flat ? `0` : `3px 0 13px -3px ${props.theme.shadow.color}`)};
  flex-shrink: 0;
`;

export const Index: React.FC<IProps> = ({className, children, ...props}) => {
  const {mobile, tablet, sidebarOpen, toggleSidebar} = useAppStore();
  return (
    <>
      <div
        className={clx('sidebarOverlay', {
          'sidebar-mobile-overlay-show': (mobile || tablet) && sidebarOpen,
        })}
        onClick={() => toggleSidebar()}
      />
      <AsideLayout {...props} className={className}>
        {children}
      </AsideLayout>
    </>
  );
};

export default Index;
