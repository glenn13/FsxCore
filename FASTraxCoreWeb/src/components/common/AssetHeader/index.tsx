import React, {HTMLAttributes, forwardRef} from 'react';

import styled from 'styled-components';

const StyledHeader = styled.h1`
  font-size: 1.25rem;
  font-weight: bold;
  padding: 0.3rem 0.6rem;
  border: 1px solid;
  border-radius: 0.4rem;
  line-height: 1.25rem;
  color: #608eb6;
`;

export interface AssetHeaderProps extends HTMLAttributes<HTMLHeadingElement> {
  children?: React.ReactNode;
  label?: string;
}

const AssetHeader = forwardRef<HTMLHeadingElement, AssetHeaderProps>(({label, ...props}, ref) => {
  return (
    <StyledHeader ref={ref} {...props}>
      {label}: {props.children}
    </StyledHeader>
  );
});
export default AssetHeader;
