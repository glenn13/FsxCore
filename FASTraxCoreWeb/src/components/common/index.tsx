export * from './Alert';
export {default as Aside} from './Aside';
export {default as AssetHeader} from './AssetHeader';
export {default as AvatarGroup} from './AvatarGroup';
export {default as AvatarUploader} from './AvatarUploader';
export {default as BannerHeader} from './BannerHeader';
export {default as Banner} from './Banner';
export {default as Block} from './Blocks';
export {default as Brand} from './Brand';
export {default as BreadcrumbItem} from './Breadcrumb/BreadcrumbItem';
export {default as Breadcrumb} from './Breadcrumb';
export {default as ButtonDropdownItem} from './ButtonDropdown/ButtonDropdownItem';
export {default as ButtonDropdown} from './ButtonDropdown';
export {default as ButtonGroup} from './ButtonGroup';
export {default as Button} from './Button';
export {default as Carousel} from './Carousel';
export {default as Chart} from './Chart';
export {default as Checkbox} from './Checkbox';
export {default as ConfirmDialog, ConfirmDialogBody, ConfirmDialogFooter} from './ConfirmDialog';
export {default as ContentSidebar} from './ContentSidebar';
export {default as Content} from './Content';
export {default as CurrencyInput} from './CurrencyInput';
export {default as FooterControl} from './FooterControl';
export {default as Footer} from './Footer';
export {default as FsxAttachmentForm} from './AttachmentForm';
export {default as FsxAttachmentField} from './AttachmentField';
export {default as FsxButton} from './FsxButton';
export {default as FsxCheckbox} from './FsxCheckbox';
export {default as FsxDatePicker} from './FsxDatePicker';
export {default as FsxDrawer} from './Drawer';
export {default as FsxDropDownList} from './FsxDropDownList';
export {default as FsxDropdown} from './Dropdown';
export {default as FsxGridColumn} from './FsxGridColumn';
export {default as FsxGridWithSelection} from './FsxGridWithSelection';
export {default as FsxGrid} from './FsxGrid';
export {default as FsxInputGroup} from './InputGroup';
export {default as FsxInput} from './FsxInput';
export {default as FsxMaskedTextBox} from './FsxMaskedTextBox';
export {default as FsxModal} from './FsxModal';
export {default as FsxNumericTextBox} from './FsxNumericTextBox';
export {default as FsxTable} from './FsxTable';
export {default as FsxTextArea} from './FsxTextArea';
export {default as FsxTextarea} from './Textarea';
export {default as FsxTimePicker} from './FsxTimePicker';
export {default as GridToolbarSearch} from './GridToolbarSearch';
export {default as LoaderOverlay} from './LoaderOverlay';
export {default as Loader} from './Loader';
export {default as NavBar} from './NavBar';
export {default as NavItem} from './NavItem';
export {default as Nav} from './Nav';
export {default as ProgressBar} from './ProgressBar';
export {default as RadialMenu} from './RadialMenu';
export {default as Radio} from './Radio';
export {default as SidebarExtension} from './SidebarExtension';
export {default as Sidebar} from './Sidebar';
export {default as SvgIcon} from './SvgIcon';
export {default as Tooltip} from './Tooltip';
export {default as TransitionWrapper} from './TransitionWrapper';
export { default as Wrapper } from './Wrapper';
export {default as Badge} from './Badge';
