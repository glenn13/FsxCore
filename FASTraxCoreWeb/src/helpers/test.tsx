import React from 'react';
import {mount, shallow} from 'enzyme';
import {render} from '@testing-library/react';
import {ThemeProvider} from 'styled-components';
import BaseTheme from '@app/theme/presets/base';

export const ThemeProviderWrapper = (theme?: any) => ({children}: any) => (
  <ThemeProvider theme={theme || BaseTheme}>{children}</ThemeProvider>
);

export const mountWithTheme = (children: any, theme?: any) =>
  mount(children, {wrappingComponent: ThemeProviderWrapper(theme)});

export const renderWithTheme = (children: any, theme?: any) =>
  render(children, {wrapper: ThemeProviderWrapper(theme)});

export const shallowWithTheme = (children: any, theme?: any) =>
  shallow(children, {wrappingComponent: ThemeProviderWrapper(theme)});
