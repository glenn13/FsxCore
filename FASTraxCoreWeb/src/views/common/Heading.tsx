import React from 'react';
import clx from 'classnames';

export interface HeadingProps {
  title: string;
  shadow?: boolean;
}

const Heading: React.FC<HeadingProps> = ({title, shadow}) => {
  return (
    <label
      className={clx(`text-base tracking-wider uppercase h-12 w-full flex items-center px-4 py-2`, {
        shadow: shadow && 'shadow z-10',
      })}
      style={{backgroundColor: '#F2F2F2'}}>
      {title}
    </label>
  );
};

export default React.memo(Heading);
