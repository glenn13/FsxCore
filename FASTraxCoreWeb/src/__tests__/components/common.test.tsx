import React from 'react';
import {render} from '@testing-library/react';
import {Button, FsxDrawer} from '@app/components/common';
import Heading from '@app/views/common/Heading';
import {renderWithTheme} from '../../helpers/test';

describe('Heading', () => {
  it('displays its title props', () => {
    const title = 'Sample Heading';

    const {getByText} = render(<Heading title={title} />);

    expect(getByText(title)).toBeInTheDocument();
  });
});

describe('Button', () => {
  it('displays its children props', () => {
    const title = 'Sample Button';
    const {getByText} = renderWithTheme(<Button>{title}</Button>);

    expect(getByText(title)).toBeInTheDocument();
  });

  it('executes onClick props', () => {
    const mockClick = jest.fn();

    renderWithTheme(<Button onClick={mockClick} />);

    document.querySelector('button')?.click();

    expect(mockClick).toBeCalled();
  });
});

describe('FsxDrawer', () => {
  it('displays its title props', () => {
    const title = 'This is Model';
    const {getByText} = renderWithTheme(<FsxDrawer title={title} />);

    expect(getByText(title)).toBeInTheDocument();
  });

  it('unMounts its children when unMountChildren property is truthy and isOpen is falsy', () => {
    const children = <div>Sample Children Div</div>;
    const firstTestId = 'FsxDrawer1';
    const firstRender = renderWithTheme(
      <FsxDrawer unMountChildren testId={firstTestId}>
        {children}
      </FsxDrawer>,
    );
    const firstElement = firstRender.getByTestId(firstTestId);

    expect(firstElement.firstChild).toBeNull();

    const secondTestId = 'FsxDrawer2';
    const secondRender = renderWithTheme(
      <FsxDrawer unMountChildren isOpen testId={secondTestId}>
        {children}
      </FsxDrawer>,
    );
    const secondElement = secondRender.getByTestId(secondTestId);

    expect(secondElement.firstChild).toMatchSnapshot(`<div>Sample Children Div</div>`);
  });

  it('displays the drawer when isOpen is true', () => {
    const {container: firstContainer} = renderWithTheme(<FsxDrawer isOpen={true} />);

    expect(firstContainer.firstElementChild?.classList.contains('visible')).toBeTruthy();
    expect(firstContainer.firstElementChild?.classList.contains('invisible')).toBeFalsy();

    const {container: secondContainer} = renderWithTheme(<FsxDrawer isOpen={false} />);

    expect(secondContainer.firstElementChild?.classList.contains('visible')).toBeFalsy();
    expect(secondContainer.firstElementChild?.classList.contains('invisible')).toBeTruthy();
  });

  it('triggers property onClose when `ams-cross` icon is clicked and when user clicks outside of the actual drawer when closeOnOutsideClick is true (default is true)', () => {
    const firstMockOnClose = jest.fn();
    const firstTestId = 'FsxDrawer1';
    const {container: firstContainer} = renderWithTheme(
      <FsxDrawer testId={firstTestId} onClose={firstMockOnClose} />,
    );
    const buttonIcon = firstContainer.querySelector('button .ams-cross')?.parentElement;

    buttonIcon?.click();

    expect(firstMockOnClose).toBeCalledTimes(1);

    const firstWrapper = firstContainer.firstElementChild as HTMLElement;
    firstWrapper.click();

    expect(firstMockOnClose).toBeCalledTimes(2);

    const content = firstWrapper.firstElementChild as HTMLElement;
    content.click();

    expect(firstMockOnClose).toBeCalledTimes(2);

    const secondMockOnClose = jest.fn();
    const secondTestId = 'FsxDrawer2';
    const {container: secondContainer} = renderWithTheme(
      <FsxDrawer testId={secondTestId} onClose={secondMockOnClose} closeOnOutsideClick={false} />,
    );
    const secondWrapper = secondContainer.firstElementChild as HTMLElement;
    secondWrapper.click();

    expect(secondMockOnClose).not.toBeCalled();
  });
});
