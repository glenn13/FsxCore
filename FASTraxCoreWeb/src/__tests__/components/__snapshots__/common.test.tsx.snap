// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FsxDrawer unMounts its children when unMountChildren property is truthy and isOpen is falsy: <div>Sample Children Div</div> 1`] = `
<div>
  Sample Children Div
</div>
`;
